grpc/options.go

```diff
@@ -0,0 +1,64 @@
+package sentrygrpc
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type options struct {
+	repanic  bool
+	reportOn ReportOn
+}
+
+func buildOptions(ff ...Option) options {
+	opts := options{
+		reportOn: ReportAlways,
+	}
+
+	for _, f := range ff {
+		f(&opts)
+	}
+
+	return opts
+}
+
+// Option configures reporting behavior.
+type Option func(*options)
```

Comment:
I wonder if we should separate options for server/client and unary/stream. Probably they won't diverge too much?

Reply:
Good point, ~~perhaps what would give us more flexibility is~~

```suggestion
type Option interface{
	apply(*options)
}
```

~~Later down the road it could become~~

```go
type Option interface{
	applyServer(*serverOptions)
	applyClient(*clientOptions)
	// ...
}
```

It would allow us to share options between servers and clients, unary and stream.
Though I'm not sure that's actually a good idea. One downside is discoverability and documentation. If server/client/etc options have the same type, all of them will be grouped together in godoc. If an option doesn't make sense to a certain type (client/server/unary/stream), that would also require documentation to explain instead of being impossible to compile because of incompatible types.

I think the options we have right now are "interceptor options", and should probably be the same for other types of interceptors? Are there other existing concepts we'd expand into?

```suggestion
type InterceptorOption interface{
	apply(*options)
}
```

If there is only reason to be one type of option, for interceptors, then we can keep the name simple, `Option`.

---

Sorry, busy day, didn't have much time to think this through but wanted to share and continue the conversation.

Reply:
I'm not really sure but both of the following changes shouldn't break user code because they don't directly use `Option`. So can we start simple?

- Changing `type Option func(*options)` to `type Option interface {}`.
- Renaming `Option` to `InterceptorOption`.

> Sorry, busy day, didn't have much time to think this through but wanted to share and continue the conversation.

No worries! We don't need to rush. I (we?) just want to agree on a good design ;)